### PR TITLE
fix(view): Update API views when removing a view

### PR DESCRIPTION
**Pull request by brasseld**
Closes gravitee-io/issues#317

### CODE REVIEW

File: gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java [hunk 1]
```diff
@@ -622,13 +622,29 @@ public ImageEntity getPicture(String apiId) {
     @Override
     public void deleteViewFromAPIs(final String viewId) {
         findAll().forEach(api -> {
-            if (api.getViews()
+            if (api.getViews() != null && api.getViews()
                     .removeIf(view -> view.equals(viewId))) {
```
Comment (aelamrani) [inline gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java]: Not sur the removeIf is necessary here..

File: gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java [hunk 2]
```diff
@@ -622,13 +622,29 @@ public ImageEntity getPicture(String apiId) {
...
-                update(api.getId(), convert(api));
+                removeView(api.getId(), viewId);
             }
         });
     }
 
+    private void removeView(String apiId, String viewId) throws TechnicalManagementException {
+        try {
+            Optional<Api> optApi = apiRepository.findById(apiId);
```
Comment (aelamrani) [inline gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java]: Not better to give the api object in parameter?

File: gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java (hunk 1 above)
Comment (brasseld) [inline gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java]: It's your stuff guy...

File: gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java (hunk 2 above)
Comment (brasseld) [inline gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java]: no

File: gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java (hunk 1 above)
Comment (aelamrani) [inline gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java]: ? I mean not anymore necessary with your refactor...

File: gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java (hunk 2 above)
Comment (aelamrani) [inline gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java]: not sure...

File: gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java (hunk 1 above)
Comment (brasseld) [inline gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java]: I don't understand why but I let you apply necessary changes.

File: gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java (hunk 2 above)
Comment (brasseld) [inline gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java]: so let's do required changes...

File: gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java (hunk 1 above)
Comment (aelamrani) [inline gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/ApiServiceImpl.java]: You are doing a remove of view twice : once by the removeIf, and on the remove() method...

**Pull request merged**